File: app/src/main/java/cn/intret/app/picgo/model/BaseModule.java
package cn.intret.app.picgo.model;

import android.content.Context;

import org.greenrobot.eventbus.EventBus;

import cn.intret.app.picgo.app.CoreModule;
import cn.intret.app.picgo.app.RxBus;

/**
 * Base service
 */

public class BaseModule {
    protected EventBus mBus;
    protected Context mContext;

    public BaseModule() {
        mContext = CoreModule.getInstance().getAppContext();
        if (CoreModule.getInstance().getAppContext() == null) {
            throw new IllegalStateException("Please initialize the CoreModule class (CoreModule.getInstance().init(appContext).");
        }
        mBus = EventBus.getDefault();
    }
}

File: app/src/main/java/cn/intret/app/picgo/model/BaseModule.kt
package cn.intret.app.picgo.model

import android.content.Context

import org.greenrobot.eventbus.EventBus

/**
 * Base service
 */

open class BaseModule {
    protected var mBus: EventBus
    protected lateinit var mContext: Context

    init {
        mBus = EventBus.getDefault()
    }

    open fun setAppContext(applicationContext: Context) {
        mContext = applicationContext
    }
}
